RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __RecoLocalCalo_HGCalRecAlgos_RecHitTools_h__
+#define __RecoLocalCalo_HGCalRecAlgos_RecHitTools_h__
+
+#include <array>
+#include <cmath>
+
+class HGCalGeometry;
+class HGCalDDDConstants;
+class DetId;
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}
+
+namespace hgcal {
+  class RecHitTools {    
+  public:
+  RecHitTools() : geom_({ {nullptr,nullptr} }), ddd_({ {nullptr,nullptr} }) {}
+    ~RecHitTools() {}
+
+    void getEvent(const edm::Event&);
+    void getEventSetup(const edm::EventSetup&);
+    
+    std::float_t getSiThickness(const DetId&) const;
```

Comment:
I don't see this type in CMSSW used much.
Behavior appears to be between float and double32 depending on the value of FLT_EVAL_METHOD.
.. I don't know what the value of FLT_EVAL_METHOD is in CMSSW distributions.

Please clarify the choice.